MVVMRxSwiftDemo/Sources/Presentation/Modules/SurveyQuestion/SurveyQuestionViewModel.swift

```diff
@@ -41,28 +45,66 @@ final class SurveyQuestionViewModel: SurveyQuestionViewModelProtocol {
     let isSubmitButtonHidden: Driver<Bool>
     let isNextButtonHidden: Driver<Bool>
     let questionIndex: Driver<String>
-   
+    let imageURL: Driver<URL?>
+    let didSubmitSurvey: Driver<Void>
+    
     private let didScrollTrigger = PublishRelay<Int>()
-    private let disposeBag = DisposeBag()
+    private let loadQuestionListTrigger = PublishRelay<Void>()
+    private let submitSurveyTrigger = PublishRelay<Void>()
     
-    init() {
-        #warning("Survey questions mock")
-        let questions = [
-            QuestionCollectionCellViewModel(
-                questionText: "How fulfilled did you feel during this WFH period?",
-                displayType: .choice,
-                pickType: .one,
-                answers: [
-                    APISurveyAnswer(id: "", displayOrder: 1, text: "Very fulfilled", inputMaskPlaceholder: nil),
-                    APISurveyAnswer(id: "", displayOrder: 2, text: "Somewhat fullfilled", inputMaskPlaceholder: nil),
-                    APISurveyAnswer(id: "", displayOrder: 3, text: "Somewhat unfulfilled", inputMaskPlaceholder: nil)
-                ]
-            )
-        ]
-        questionItems = Driver.just(questions)
-        isSubmitButtonHidden = didScrollTrigger.map { $0 != questions.count }.asDriverOrEmptyIfError()
-        isNextButtonHidden = didScrollTrigger.map { $0 == questions.count }.asDriverOrEmptyIfError()
-        questionIndex = didScrollTrigger.map { "\($0)/\(questions.count)" }.asDriverOrEmptyIfError()
+    init(factory: ModuleFactoryProtocol, surveyId: String) {
+        let disposeBag = DisposeBag()
+        let questionSubmissions = BehaviorRelay<[QuestionSubmission]>(value: [])
+
+        let questions: Driver<[SurveyQuestion]> = loadQuestionListTrigger
+            .flatMapLatest {
+                factory
+                    .getSurveyDetailUseCase()
+                    .getSurveyDetail(id: surveyId)
+                    .map { survey -> [SurveyQuestion] in
+                        survey.sortedQuestions?
+                            .filter { $0.displayType != .intro && $0.displayType != .outro } ?? []
+                    }
+            }
+            .asDriver(onErrorJustReturn: [])
```

Comment:
Please move this out of the `.flatMapLatest` block for easy to see the steps:

```suggestion
            .flatMapLatest { // get data from API
                factory
                    .getSurveyDetailUseCase()
                    .getSurveyDetail(id: surveyId)
            }
            .compactMap { survey -> [SurveyQuestion] in // process data and use compactMap to ignore nil data
                survey.sortedQuestions?.filter { $0.displayType != .intro && $0.displayType != .outro }
            }
            .asDriver(onErrorJustReturn: [])
```